### ISSUE TITLE
content-sqlite may leave on-disk data inconsistent on a crash

### ISSUE THREAD

**Issue by garlick** [flux-framework/flux-core#4152]
Problem: The `content-sqlite` content backing store module sets [SQLite pragmas](https://www.sqlite.org/pragma.html) that improve performance, but relax consistency guarantees of SQL commits, like `journal_mode=OFF` and `synchronous=OFF`.  Consequently, the backing store could become corrupted after a rank 0 crash, complicating recovery.

We should consider an option (especially for system instance) that disables these performance hacks.

After quantifying the before and after performance, we may want to investigate some mitigations.  For example, right now each blob is committed separately.  We could add multiple blobs to a commit, then after a threshold size, or a timer expiration, commit them and send the responses for all once the commit finishes.

**Comment by garlick** [comment]
An example:
```
Apr 03 13:07:06 picl0 flux[14320]: content-sqlite.err[0]: setting sqlite 'journal_mode' pragma: database disk image is malformed(11)
```

**Comment by garlick** [comment]
Another example.  This time the instance was shut down cleanly with `sudo flux shutdown`.  I'm not sure what caused the database to become malformed.
```
Apr 16 16:22:30 picl0 flux[86356]: broker.info[0]: shutdown: run->cleanup 38.429m
Apr 16 16:22:30 picl0 flux[86356]: broker.info[0]: cleanup.0: flux queue stop --quiet Exited (rc=0) 0.0s
Apr 16 16:22:30 picl0 flux[86356]: broker.info[0]: cleanup.1: flux job cancelall --user=all --quiet -f --states RUN Exited (rc=0) 0.0s
Apr 16 16:22:30 picl0 flux[86356]: broker.info[0]: cleanup.2: flux queue idle --quiet Exited (rc=0) 0.0s
Apr 16 16:22:30 picl0 flux[86356]: broker.info[0]: cleanup-success: cleanup->shutdown 0.12144s
Apr 16 16:22:30 picl0 flux[86356]: broker.info[0]: children-none: shutdown->finalize 0.437481ms
Apr 16 16:22:30 picl0 flux[86356]: broker.info[0]: rc3.0: /usr/local/etc/flux/rc3 Exited (rc=0) 0.3s
Apr 16 16:22:30 picl0 flux[86356]: broker.info[0]: rc3-success: finalize->goodbye 0.338493s
Apr 16 16:22:30 picl0 flux[86356]: broker.info[0]: goodbye: goodbye->exit 1.144ms
Apr 16 16:22:30 picl0 systemd[1]: flux.service: Main process exited, code=exited, status=42/n/a
Apr 16 16:22:30 picl0 systemd[1]: flux.service: Failed with result 'exit-code'.
Apr 16 16:22:30 picl0 systemd[1]: flux.service: Consumed 27min 7.322s CPU time.
-- Boot f29bb2a634904eff95dac0f3ed89a8dd --
Apr 16 16:22:39 picl0 systemd[1]: Starting Flux message broker...
Apr 16 17:53:11 picl0 systemd[1]: Started Flux message broker.
Apr 16 17:53:12 picl0 flux[734]: broker.info[0]: start: none->join 57.096ms
Apr 16 17:53:12 picl0 flux[734]: broker.info[0]: parent-none: join->init 0.079759ms
Apr 16 17:53:12 picl0 flux[734]: broker.info[0]: online:  (ranks )
Apr 16 17:53:12 picl0 flux[734]: content-sqlite.err[0]: setting sqlite 'journal_mode' pragma: database disk image is malformed(11)
Apr 16 17:53:12 picl0 flux[734]: content-sqlite.crit[0]: module exiting abnormally
```

**Comment by garlick** [comment]
Should have been closed by #4294 I think.
